### PR TITLE
Chore: Update check_license script to update copyright year

### PR DESCRIPTION
**Pull request by onetechnical**
## Summary

Update `scripts/check_license.sh` to add an update option. Also added an EXTRA_FILES array to add non-golang files that also have copyrights.

This script should then be used to create a separate PR with updates for the 1000+ files it affects.

## Test Plan

- Verified that `./scripts/check_license.sh` errored.
- Verified output of `./scripts/check_license.sh -v`
- Verified modifications of files when running `./scripts/check_license.sh -u`
- Verified `./scripts/check_license.sh` passes after modifications

### CODE REVIEW

**Review by jasonpaulos — APPROVED** [review]
Upgrade flag is a good idea

**Comment by jasonpaulos** [comment]
Actually I think this also needs to be updated: https://github.com/algorand/go-algorand/blob/35bcd35ad09dbba7c4d07c7cbf04510f55132aab/cmd/tealdbg/bundle_home_html.sh#L6

File: scripts/check_license.sh [hunk 1]
```diff
@@ -62,7 +68,11 @@ for FILE in $VERSIONED_GO_FILES; do
             RETURN_VALUE=1
 
             if ! $VERBOSE; then
-                if $INPLACE; then
+                if $UPGRADE; then
+                    sed -i.orig s/Copyright\ \(C\)\ 2019-....\ Algorand,\ Inc\./Copyright\ \(C\)\ 2019-$CURRENT_YEAR\ Algorand,\ Inc./ "$PROJECT_ROOT/$FILE" && \
```
Comment (algorandskiy) [inline scripts/check_license.sh:72]: 1. maybe replace `....` with PREV_YEAR  to be exact?
2. if I recall right `sed -i.orig` is mac only syntax (btw, `sed -i '' -e` also works without creating extra files). I think this cmd should be BSD/GNU sed universal.

**Review by algorandskiy — COMMENTED** [review]
I suggest to extract `scripts/check_license.sh` changed from this PR to 1) simplify review and 2) do not hold the license header update while working on `check_license.sh` changes.

File: scripts/check_license.sh (hunk 1 above)
Comment (onetechnical) [reply scripts/check_license.sh:85]: I'd keep it general to catch missing years, but don't feel strongly on it.

**Review by algorandskiy — APPROVED** [review]

File: scripts/check_license.sh [hunk 2]
```diff
@@ -77,17 +100,48 @@ for FILE in $VERSIONED_GO_FILES; do
     fi
 done
 
+# non-go files that include a license header
+for FILE in "${EXTRA_FILES[@]}"; do
+    if ! grep -qs "Copyright (C) 2019-$CURRENT_YEAR Algorand, Inc." "$PROJECT_ROOT/$FILE"; then
+        RETURN_VALUE=1
```
Comment (jasonpaulos) [inline scripts/check_license.sh:106]: minor: it seems like it should not set the return value to 1 if this is updating the files?

**Review by jasonpaulos — COMMENTED** [review]
Probably a good idea to make the script changes in a separate PR.

Could you update the title as well? It still implies this PR will fix the license dates

File: scripts/check_license.sh (hunk 2 above)
Comment (onetechnical) [reply scripts/check_license.sh:106]: As currently structured, it returns 1 if anything changes. I suppose a slight improvement might be to change all updates and in place edits to not set this and only print if $MOD_COUNT > 0, but I'm not sure we gain much by it.

Comment (jasonpaulos) [reply scripts/check_license.sh:106]: Ok, I didn't realize that. This makes sense

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**